applications/nrf5340_audio/src/audio_module/src/amod_api.h

```diff
@@ -0,0 +1,492 @@
+
+/**
+ * @brief Stop a module processing data.
+ *
+ * @param handle  The handle for the module to be stopped
+ *
+ * @return 0 if successful, error value
+ */
+int amod_stop(struct amod_handle *handle);
+
+/**
+ * @brief Send a data buffer to a module, all data is consumed by the module.
+ *
+ * @param handle       The handle for the receiving module instance
+ * @param block       Pointer to the audio data block to send to the module
+ * @param response_cb  A pointer to a callback to run when the buffer is
+ *                     fully comsumed in a module
+ *
+ * @return 0 if successful, error value
+ */
+int amod_data_tx(struct amod_handle *handle, struct aobj_block *block,
+		 amod_response_cb response_cb);
+
+/**
+ * @brief Retrieve data from the module.
+ *
+ * @param handle  A handle to this module instance
+ * @param block  Pointer to the audio data block from the module
+ * @param  timeout    Non-negative waiting period to wait for operation to complete
+ *	                  (in milliseconds). Use K_NO_WAIT to return without waiting,
+ *	                  or K_FOREVER to wait as long as necessary.
+ *
+ * @return 0 if successful, error value
+ */
+int amod_data_rx(struct amod_handle *handle, struct aobj_block *block, k_timeout_t timeout);
+
+/**
+ * @brief Send and retrieve data from the module all data is consumed within
+ *        the call. The input data buffer maybe released once function retrurned.
+ *
+ * @param handle_tx   A handle to the module to send the input block to
+ * @param handle_rx   A handle to the module to receive an block from
+ * @param block_tx   Pointer to the audio data block to send
+ * @param block_rx   Pointer to the audio data block received
+ * @param timeout     Non-negative waiting period to wait for operation to complete
+ *	                  (in milliseconds). Use K_NO_WAIT to return without waiting,
+ *	                  or K_FOREVER to wait as long as necessary.
+ *
+ * @return 0 if successful, error value
+ */
+int amod_data_tx_rx(struct amod_handle handle_tx, struct amod_handle handle_rx,
+		    struct aobj_block *block_tx, struct aobj_block *block_rx, k_timeout_t timeout);
+
+/**
+ * @brief Helper function to configure the thread information for the module
+ *        set-up parameters structure.
+ *
+ * @param parameters   Pointer to the module set-up parameters
+ * @param description  Pointer to the modules private description
+ * @param stack        Memory block for the threads stack
+ * @param stack_size   Size of the threads stack
+ * @param priority     Priority of the thread insatnce, one of #amod_id
+ * @param msg_rx       Number of concurrent input messages
+ * @param msg_tx       Number of concurrent output messages
+ * @param data_slab    Slab to take a data item from
+ * @param data_size    Size of the data item that will be taken
+ *
+ * @return 0 if successful, error value
+ */
+int amod_parameters_configure(struct amod_parameters *parameters,
```

Comment:
Why not just send a pointer to a amod_thread_configuration?

Reply:
The comment is incorrect it should read:

* @brief Helper function to configure the module's parameter structure with the module's description and thread information.